### PR TITLE
Use clspv api through Cmake option

### PR DESCRIPTION
**Pull request by alan-baker**
Adds Cmake option to select compile style:
 * default remains separate process
 * new option uses Clspv API

Changes:
* Changes to support using Clspv API
* update UBO test to meet std140 rules

I've tested the simple tests using both compiler options.

### CODE REVIEW

File: tests/simple_ubo/simple.cpp [hunk 1]
```diff
@@ -129,7 +129,8 @@ int main(int argc, char* argv[])
     bool success = true;
     auto buffer_data = static_cast<cl_int*>(ptr);
     for (cl_uint i = 0; i < BUFFER_SIZE/sizeof(cl_int); ++i) {
-        if (buffer_data[i] != static_cast<cl_int>(i + 1)) {
+        int i_div_4 = i / 4;
+        if (buffer_data[i] != static_cast<cl_int>(i_div_4 + 1)) {
             printf("Failed comparison at buffer_data[%d]: expected %d but got %d\n",
                    i, i + 1, buffer_data[i]);
```
Comment (kpet) [inline tests/simple_ubo/simple.cpp]: Didn't spot you forgot to update the message, I'll fix that as part of folding this test into the framework.

File: tests/simple_ubo/simple.cpp [hunk 2]
```diff
@@ -71,7 +71,7 @@ int main(int argc, char* argv[])
     CHECK_CL_ERRCODE(err);
 
     // Build program
-    err = clBuildProgram(program, 1, &device, " -constant-args-ubo -inline-entry-points ", nullptr, nullptr);
+    err = clBuildProgram(program, 1, &device, " -constant-args-ubo -inline-entry-points -relaxed-ubo-layout ", nullptr, nullptr);
```
Comment (kpet) [inline tests/simple_ubo/simple.cpp]: `-relaxed-ubo-layout` isn't really needed, is it?

Comment (alan-baker) [reply tests/simple_ubo/simple.cpp]: No that was in there from an earlier debugging effort. I forgot to remove it.

Comment (kpet) [reply tests/simple_ubo/simple.cpp]: Thanks for confirming.

**Pull request merged**